transformations/transformations/pool_allocator.py

```diff
@@ -345,12 +358,24 @@ def _get_stack_storage_and_size_var(self, routine, stack_size):
         if variables_append:
             routine.variables += as_tuple(variables_append)
         if body_prepend:
-            routine.body.prepend(body_prepend)
+            if not self._insert_stack_at_loki_pragma(routine, body_prepend):
```

Comment:
Why is there a requirement for this pragma? Can the injection point lookup be improved instead?